Tools/scripts/check_shared_ext.py

```diff
@@ -0,0 +1,200 @@
+"""Check shared extension modules
+"""
+
+import argparse
+import collections
+import enum
+import importlib._bootstrap
+from importlib.machinery import ExtensionFileLoader, BuiltinImporter
+from importlib.util import spec_from_file_location, spec_from_loader
+import os
+import pathlib
+import logging
+import sys
+import sysconfig
+import warnings
+
+
+PYBUILDDIR_TXT = "pybuilddir.txt"
+EXT_SUFFIX = sysconfig.get_config_var("EXT_SUFFIX")
+CROSS_COMPILING = "_PYTHON_HOST_PLATFORM" in os.environ
+
+
+logger = logging.getLogger(__name__)
+
+parser = argparse.ArgumentParser(
+    prog="check_shared_ext", description="Check shared extension modules"
```

Comment:
I think, more specific wording like this would be better:

```suggestion
    description="Check if shared extension modules are importable",
```